torch/utils/data/dataloader.py

```diff
@@ -1043,12 +1100,16 @@ def _process_data(self, data):
             data.reraise()
         return data
 
-    def _shutdown_worker(self, worker_id):
-        # Mark a worker as having finished its work and dead, e.g., due to
+    def _mark_worker_as_unavailable(self, worker_id, shutdown=False):
+        # Mark a worker as having finished its work e.g., due to
         # exhausting an `IterableDataset`. This should be used only when this
         # `_MultiProcessingDataLoaderIter` is going to continue running.
 
-        assert self._workers_status[worker_id]
+        # this method will lead to the worker process ending if _workers_done_event
+        # is set, but will only mark it as unavailable for this epoch otherwise.
```

Comment:
I don't think that this is true anymore.  Even when the case of `persistent_worker = True`, this is still called just once per worker, right?

Reply:
Exactly, this is called just once per worker, this is a stale comment, thanks a lot for the careful review 🙇‍♂️